staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields_test.go

```diff
@@ -141,3 +152,38 @@ func TestRoundTripManagedFields(t *testing.T) {
 		})
 	}
 }
+
+func TestDecodeManager(t *testing.T) {
+	tests := []struct {
+		managedFieldsEntry string
+		expected           string
+	}{
+		{
+			managedFieldsEntry: `
+apiVersion: v1
+fields:
+  f:apiVersion: {}
+manager: foo
+operation: Update
+time: 1:00pm
+`,
+			expected: "{\"manager\":\"foo\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"1:00pm\"}",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.managedFieldsEntry, func(t *testing.T) {
+			var unmarshaled metav1.ManagedFieldsEntry
+			if err := yaml.Unmarshal([]byte(test.managedFieldsEntry), &unmarshaled); err != nil {
+				t.Fatalf("did not expect yaml unmarshalling error but got: %v", err)
+			}
+			decoded, err := decodeManager(&unmarshaled)
```

Comment:
```suggestion
			decoded, err := DecodeManager(&unmarshaled)
```

Reply:
Thanks! I think I also need to update something in patchhandler_test